File: agent-guard-plugin/lib/events.js
const fs = require("node:fs");
const { eventsPath } = require("./state");

function appendEvent(rootDir, event) {
  const enriched = {
    ts: new Date().toISOString(),
    ...event,
  };
  fs.appendFileSync(eventsPath(rootDir), `${JSON.stringify(enriched)}\n`);
  return enriched;
}

module.exports = {
  appendEvent,
};

File: agent-guard-plugin/src/agent_guard/events.py
"""Event log helpers for workflow transitions and runtime activity."""
from __future__ import annotations

import json
from datetime import datetime, timezone
from pathlib import Path
from typing import Any

from .state import events_path


def append_event(root_dir: Path, event: dict[str, Any]) -> dict[str, Any]:
    """Append event."""
    enriched = {"ts": datetime.now(timezone.utc).isoformat(), **event}
    with events_path(root_dir).open("a", encoding="utf-8") as handle:
        handle.write(json.dumps(enriched) + "\n")
    return enriched
